x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/InferenceActionResponseTests.java

```diff
@@ -46,7 +46,7 @@ protected Writeable.Reader<InferenceAction.Response> instanceReader() {
     @Override
     protected InferenceAction.Response createTestInstance() {
         var result = switch (randomIntBetween(0, 2)) {
-            case 0 -> TextEmbeddingResultsTests.createRandomResults();
+            case 0 -> TextEmbeddingResultsTests.createRandomFloatResults();
```

Comment:
@davidkyle just a heads up. I was getting some bwc failures which I think are because of the `TextEmbeddingResults` conversion to namedwritables. So I'm forcing this to only use floats. Maybe there's an error in the way I've written that logic?